docs/arch/adr-010-new-versioning-scheme.md

```diff
@@ -0,0 +1,193 @@
+# ADR 10: New kURL Versioning Scheme
+
+## Context
+
+Based on our prior experience, offering customers the ability to select various add-on versions provides them with significant flexibility, but it also creates numerous challenges. Presently, we're struggling to ensure that each customer's chosen add-ons are compatible with one another and that migration paths are available for all potential combinations. Moreover our user feedback indicates that stability and predictability are their top priorities, rather than the specific technologies they are running behind the scenes.
+
+## Decision
+
+Based on users feedback and the challenges we are facing to maintain kURL upgrades predictable, we have decided to adopt a more opinionated approach to delivering our Kubernetes distribution. This will involve implementing a well-designed versioning scheme, which is outlined in detail in this Architecture Decision Record.
+
+## Solution
+
+The solution involves three different entities named _Channels_, _Installers_ and _Bundles_, each one of them are described below.
+
+### Choosing the Channel
+
+Our revised versioning scheme entails organising our offering into Channels that align with specific Kubernetes minor versions. As a result, users will be required to select the Channel to which they want their installer assigned.
+
+Although we won't initially provide a Channel for every Kubernetes minor version, we have arbitrarily selected Kubernetes minor version 1.24 as our starting point. At the time of writing, here are the Channels that we will be offering to our users:
+
+| Channel  | Kubernetes version installed by this Channel    |
+|----------|-------------------------------------------------|
+| v1.24    | Most up-to-date Kubernetes version for v1.24.x. |
+| v1.25    | Most up-to-date Kubernetes version for v1.25.x. |
+| v1.26    | Most up-to-date Kubernetes version for v1.26.x. |
+| v1.27    | Most up-to-date Kubernetes version for v1.27.x. |
+
+By drafting an installation file similar to the model given below, users can link their setup with a particular kURL Channel (_therefore they will be adhering to Kubernetes versions 1.27_)
+```yaml
+apiVersion: "cluster.kurl.sh/v1beta2"
+kind: "Installer"
+metadata:
+  name: "cluster"
+spec:
+  channel:  "1.27"
```

Comment:
What channel means for kURL? 
Channel seems to be a valid term for the vendor portal and not kURL.
Also, I understand that we are here trying to create an spec to indicates what is the k8s version which should be installed and managed by kurl therefore, why not make that clear by using "`kubernetes: 1.27`"

Reply:
I kind of like the naming of channel. It helps people better understand that the "kurl" portion of the version is the "minor" version and the "channel" is the major version. As soon as we started talking about these as channels internally it cleared up a lot of confusion. Carrying that same phrasing through the rest of kURL is surly going to help others too.

Reply:
We have vendor Channels I think it started to bring misleading.

Reply:
I have updated the ADR to use "Kubernetes" instead of "Channel", please take a look.

Reply:
I dont think we have settled yet on Kubernetes vs Channel. No need to revert back but lets continue to discuss

Reply:
After our last meeting we have decided to adopt "Channel".